src/function.rs

```diff
@@ -91,52 +90,71 @@ impl<'cb> ReturnValue<'cb> {
 /// including the receiver, the number and values of arguments, and
 /// the holder of the function.
 #[repr(C)]
-pub struct FunctionCallbackInfo(Opaque);
-
-impl InIsolate for FunctionCallbackInfo {
-  fn isolate(&mut self) -> &mut Isolate {
-    self.get_isolate()
-  }
+pub struct FunctionCallbackInfo {
+  implicit_args: *mut Opaque,
+  values: *mut Value,
+  length: int,
 }
 
-impl<'s> ToLocal<'s> for FunctionCallbackInfo {}
+pub struct FunctionCallbackArguments<'s> {
+  info: *const FunctionCallbackInfo,
+  phantom: PhantomData<&'s ()>,
+}
 
-impl FunctionCallbackInfo {
-  /// The ReturnValue for the call.
-  pub fn get_return_value(&self) -> ReturnValue {
-    let mut rv = MaybeUninit::<ReturnValue>::uninit();
-    unsafe {
-      v8__FunctionCallbackInfo__GetReturnValue(self, rv.as_mut_ptr());
-      rv.assume_init()
+impl<'s> FunctionCallbackArguments<'s> {
+  fn from_function_callback_info(info: *const FunctionCallbackInfo) -> Self {
+    Self {
+      info,
+      phantom: PhantomData,
     }
   }
 
-  /// The current Isolate.
-  #[allow(clippy::mut_from_ref)]
-  pub fn get_isolate(&mut self) -> &mut Isolate {
-    unsafe { v8__FunctionCallbackInfo__GetIsolate(self) }
-  }
-
-  /// The number of available arguments.
   pub fn length(&self) -> int {
-    unsafe { v8__FunctionCallbackInfo__Length(self) }
+    unsafe {
+      let length = (*self.info).length;
+      debug_assert_eq!(length, v8__FunctionCallbackInfo__Length(self.info));
```

Comment:
This check probably will avoid the problem mentioned above. I would still suggest a warning comment on FunctionCallbackInfo's declaration.